Yank/repex.py

```diff
@@ -1,752 +1,1428 @@
-
-TODO
-
-* Add analysis facility accessible by user.
-* Give up on Context caching and revert to serial Context creation/destruction if we run out of GPU memory (issuing an alert).
-* Store replica self-energies and/or -ln q(x) for simulation (for analyzing correlation times).
-* Add analysis facility.
-* Allow user to call initialize() externally and get the NetCDF file handle to add additional data?
-* Store / restore parameters and System objects from NetCDF file for resuming and later analysis.
-* Sampling support:
-  * Short-term: Add support for user to specify a callback to create the Integrator to use ('integrator_factory' or 'integrator_callback').
-  * Longer-term: Allow a more complex MCMC sampling scheme consisting of one or more moves to be specified through mcmc.py facility.
-* Allow different choices of temperature handling during exchange attempts:
-  * scale velocities (exchanging only on potential energies) - make this the default?
-  * randomize velocities (exchanging only on potential energies)
-  * exchange on total energies, preserving velocities (requires more replicas)
-* Add control over number of times swaps are attempted when mixing replicas, or compute best guess automatically
-* Add another layer of abstraction so that the base class uses generic log probabilities, rather than reduced potentials?
-* Use interface-based checking of arguments so that different implementations of the OpenMM API (such as pyopenmm) can be used.
-* Eliminate file closures in favor of syncs to avoid closing temporary files in the middle of a run.
+- ReplicaExchange
+    Base class for general replica-exchange simulations.
+- ParallelTempering
+    Convenience subclass of ReplicaExchange for parallel tempering simulations
+    (one System object, many temperatures).
 
 COPYRIGHT
 
-Written by John D. Chodera <jchodera@gmail.com> while at the University of California Berkeley.
+Written by John D. Chodera <jchodera@gmail.com> while at the University of
+California Berkeley.
 
 LICENSE
 
 This code is licensed under the latest available version of the MIT License.
 
 """
 
-#=============================================================================================
+# ==============================================================================
 # GLOBAL IMPORTS
-#=============================================================================================
+# ==============================================================================
 
 from simtk import openmm
 from simtk import unit
 
-import os, os.path
+import os
 import math
 import copy
 import time
+import inspect
 import datetime
 import logging
+import importlib
+import collections
 
 import numpy as np
 import mdtraj as md
 import netCDF4 as netcdf
 
-from .utils import is_terminal_verbose, delayed_termination
+import openmmtools as mmtools
+from openmmtools.states import ThermodynamicState
+
+from yank import utils, mpi, version
```

Comment:
I thought relative imports were discouraged. Are you sure this is a problem? I haven't had issues with Python 3 so far with absolute imports.